app/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataRecord.java

```diff
@@ -307,7 +307,8 @@ public void setUserDataGroups(Set<String> userDataGroups) {
     /** @see #getUserSubstudyMemberships() */
     @Override
     public void setUserSubstudyMemberships(Map<String, String> userSubstudyMemberships) {
-        this.userSubstudyMemberships = userSubstudyMemberships;
+        this.userSubstudyMemberships = (userSubstudyMemberships != null && !userSubstudyMemberships.isEmpty())
+                ? userSubstudyMemberships : null;
```

Comment:
Should we add a test to HealthDataRecordTest similar to emptyDataGroupSetConvertedToNull()?

Reply:
Updated